chain-impl-mockchain/src/vote/manager.rs

```diff
@@ -344,19 +367,24 @@ impl ProposalManager {
 
         let mut result = TallyResult::new(self.options.clone());
         for (choice, &weight) in decrypted_proposal.tally_result.iter().enumerate() {
-            result.add_vote(Choice::new(u8::try_from(choice).unwrap()), weight)?;
+            result
+                .add_vote(Choice::new(u8::try_from(choice).unwrap()), weight)
+                .map_err(|error| match error {
+                    VoteError::InvalidChoice { options, choice } => {
```

Comment:
How can this actually happen? We build the tally with the correct number of options ourselves

Reply:
Well, I think it can't, `Tally::verify` should take care of validating that, I think?